runtime/native.go

```diff
@@ -337,7 +337,27 @@ func getNativeType(rtype reflect.Type) *Type {
 				d[meth.Name] = newNativeMethod(meth.Name, meth.Func)
 			}
 		}
-		t = &newNativeType(rtype, base, newStringDict(d)).Type
+
+		// Building a field for this type depends on the type itself, but we
+		// can't construct the type with the real `*Dict` yet, because we
+		// couldn't otherwise add the new fields to the `*Dict` (because we
+		// don't have access to a `*Frame` in this function). Instead, we'll
+		// convert `d` into a `*Dict` after we construct all of its fields, and
+		// we'll stick it back on `t` at that point.
+		t = &newNativeType(rtype, base, nil).Type
```

Comment:
You can remove the dict param from newNativeType(). The function is used in one other spot in native_test.go and I think you'll have to manually populate the dict field on the result.